tests/test_zppy_e3sm_diags.py

```diff
@@ -533,6 +533,32 @@ def test_add_climo_dependencies(self):
         dependencies = []
         add_climo_dependencies(c, dependencies, "script_dir")
         self.assertEqual(dependencies, ["script_dir/climo_cdsub_1980-1990.status"])
+        c = {"sets": ["diurnal_cycle"]}
+        c.update(base)
+        dependencies = []
+        self.assertRaises(
+            ParameterNotProvidedError,
+            add_climo_dependencies,
+            c,
+            dependencies,
+            "script_dir",
+        )
```

Comment:
Test that `climo_diurnal_subsection` is checked for if `diurnal_cycle` is requested.

(In this unit test, we have `"guess_path_parameters": True, "guess_section_parameters": True`, so an error will be raised if this parameter is not provided. These options are off-by-default, meaning `zppy` will instead try to guess the value.)